python/tk_multi_workfiles/file_filters.py

```diff
@@ -1,86 +1,109 @@
-        self._users = [g_user_cache.current_user] if g_user_cache.current_user else []
+        self._reset_user_lists()
 
-    #@property
+    # @property
     def _get_show_all_versions(self):
         return self._show_all_versions
-    #@show_all_versions.setter
+
+    # @show_all_versions.setter
     def _set_show_all_versions(self, show):
         if show != self._show_all_versions:
             self._show_all_versions = show
             self.changed.emit()
-    show_all_versions=property(_get_show_all_versions, _set_show_all_versions)
 
-    #@property filter_reg_exp
+    show_all_versions = property(_get_show_all_versions, _set_show_all_versions)
+
+    # @property filter_reg_exp
     def _get_filter_reg_exp(self):
         return self._filter_reg_exp
-    #@filter_reg_exp.setter
+
+    # @filter_reg_exp.setter
     def _set_filter_reg_exp(self, value):
         if value != self._filter_reg_exp:
             self._filter_reg_exp = value
             self.changed.emit()
+
     filter_reg_exp = property(_get_filter_reg_exp, _set_filter_reg_exp)
 
-    #@property
-    def _get_available_users(self):
+    @property
+    def available_users(self):
+        """
+        :returns: List of available user sandboxes.
+        """
         return self._available_users
-    #available_users.setter
-    def _set_available_users(self, users):
-        current_user_ids = set([u["id"] for u in self._available_users if u])
-        new_user_ids = set([u["id"] for u in users if u])
-        if new_user_ids != current_user_ids:
-            self._available_users = [u for u in users if u]
-            self.available_users_changed.emit(self._available_users)
 
-            # also, update user ids, removing any that are no longer in
-            # the list of available user ids:
-            self._set_users(self._users)
-    available_users = property(_get_available_users, _set_available_users)
+    def clear_available_users(self):
+        """
+        Clear the list of available user sandboxes.
+        """
+        self._reset_user_lists()
+        self.available_users_changed.emit(self._available_users)
 
-    #@property
+    def _reset_user_lists(self):
+        self._available_users = [g_user_cache.current_user] if g_user_cache.current_user else []
+        self._users = [g_user_cache.current_user] if g_user_cache.current_user else []
+
+    def add_users(self, users):
+        """
+        Adds to the list of available user sandboxes.
+        """
+        nb_users_before = len(self._available_users)
+
+        # merge the two lists, discarding doubles.
+        new_users_by_id = {user["id"]: user for user in users}
+        available_users_by_id = {user["id"]: user for user in self._available_users}
+        available_users_by_id.update(new_users_by_id)
+        self._available_users = available_users_by_id.values()
+
+        # Something was added to the set.
+        if nb_users_before != len(self._available_users):
```

Comment:
I don't think I know enough about how workfiles works to understand this scenario completely. Is this always an additive thing? i.e., could the number be the same and the users be different?